upgraderesponder/db-cache.go

```diff
@@ -8,7 +8,7 @@ import (
 	"time"
 )
 
-const maxSyncRetries = 5
+const maxSyncRetries = 2
```

Comment:
Doesn't adding the client timeout solve the issue? Could you explain the purpose of decreasing the retries?

Reply:
If we don't decrease maxSyncRetries. It might take 50 seconds to get out of the sync() and release the lock. During this time, a lot of client (longhorn-manager) might timeout and cannot get update. 

I am thinking 10s * 2 is reasonable time. If it takes longer, the server may better just drop the current batch and handle the incoming request.

Reply:
I see, I don't have a good figure for the retries.

Just curious, after releasing the lock after x seconds, the timed-out client will attempt to get sync and update again later?

Reply:
I think if x is smaller than the client timeout, then the client request will just succeed. Looks like 20s is smaller than the client timeout

Reply:
> Just curious, after releasing the lock after x seconds, the timed-out client will attempt to get sync and update again later

When the client timed out, I think it depends on the implementation of the client, they might retry or just ignore the error.

Reply:
Thank you for explaining. To summarize, we are reducing the retries to minimize the number of clients that would be impacted by the timeout.